### DISCUSSION TITLE
Oracle target - specify primary key, unique and indexes - table_keys does not seem to be used

### DISCUSSION THREAD

**Discussion by fabrice-etanchaud** [slingdata-io/sling-cli#225] [Q&A] [answered]
Hi !
Reading all the work done with table_keys for starrocks, I tried to use table_keys to create primary key or unique or indexes in target oracle tables, with no luck.
Am I missing something ? 
Is there another way to do it ?
Best regards,
Fabrice

**Comment by flarco** [comment]
Good catch, it's not implemented yet. `table_keys` only works for starrocks at the moment.
I'll add others soon.

**Reply by fabrice-etanchaud** [reply]
Thank you so much, @flarco !
I thought about using post_sql but that should be done only when mode=full-refresh.

Best regards from French west coast.

**Comment by GoranDervisevic** [comment]
Hi!
We are using Sling to load data from an Oracle database to an SQL Server database and I was having a discussion with my colleague if Sling is capable to add index automatically once a table is loaded. And I came upon this discussion and the use of table_keys which should the same purpose. Correct?

@flarco has table_keys been implemented for other db or SQL server? Thank you!

**Comment by flarco** [comment]
@GoranDervisevic Yes: https://github.com/slingdata-io/sling-cli/issues/226

**Comment by GoranDervisevic** [comment]
Thank you for the quick response @flarco and have a question that I should have asked in the first place. I could not find any answers or method on the official Sling website,

How is table_keys implemented in Sling? Is it part of a Sling flag when running?

**Comment by flarco** [comment, marked as answer]
In a replication:

```yaml
streams:
  my_stream:
    target_options:
      table_keys:
        index: [col1, col2]
```

**Reply by GoranDervisevic** [reply]
Thank you! And my mistake, table_keys description is in the official Sling website documentation. I'm scrolling too fast :)

https://docs.slingdata.io/sling-cli/run/configuration#target
